Document the BrowserType class in its own page and document the top level module properties on the landing page (#788)
grafana/k6-docs: src/data/markdown/docs/30 xk6-browser/01 xk6-browser/03 BrowserType.md

---
title: "BrowserType"
excerpt: "xk6-browser: BrowserType Class"
---

> The `BrowserType` is the entry point into launching a browser process; `chromium` is currently the only supported `BrowserType`. To use it, import `chromium` from the top level module `/k6/x/browser`.

Reviewer: ```suggestion
The `BrowserType` is the entry point into launching a browser process; `chromium` is currently the only supported `BrowserType`. To use it, import `chromium` from the top level module `k6/x/browser`.
```

Reviewer: Hey, I still see `/k6/x/browser` on b01db6f2.

Author: Hmm, it looks like i added it back in 🤔, i've removed it now in cbdbfaa2b28fffbab81284e72b58b93df165c449.

| Method            | Description                                                                                                      |
|-------------------|------------------------------------------------------------------------------------------------------------------|

> | launch([options]) | Launches a new browser process. You can see a list of all possible `options` in [the options heading](#options). |

Reviewer: Do we want to document the [other methods](https://github.com/grafana/xk6-browser/blob/c5c2cead896d615f06f70f0f4ed24e1cd041b1ed/api/browser_type.go#L28-L34) as well?

`connect()` and `launchPersistentContext()` aren't implemented yet, but the others would be good to have. We should list the not-implemented ones here as well with a WIP icon, but we can leave out documenting their options, since they're not usable.

It would be nice if we could use the new collapsible element for documenting these options as well, so we don't have to use a separate table for it, though it probably wouldn't make sense, so nevermind. Let's just change the header below to "`launch()` options", so that it's clear. And then change this to:

```suggestion
| launch([options]) | Launches a new browser process. You can see a list of all possible `options` in [the table below](#launch-options). |
```

(Confirm that the link actually works... I'm not sure if our Gatsby JS will properly convert "`launch()` options" to `#launch-options`.)

Author: Yeah, i forgot about those. I've added the missing methods. I've also moved the options for `launch` into its own page, just like we're doing with the other classes. This also means we can use the nested table for the options.

Resolved in: 805dd25465b382f0e5edc1797adad4ac3000d8a1

Reviewer: Nice, though I'm not sure if it makes sense to nest the options in that case. If they're on a separate page in their own table already, they can be shown expanded. We can use the collapsible element for nested sub-properties, though.

Author: I've removed the nesting for those options.

> ### Options

Reviewer: To clarify:
```suggestion
### `launch()` options
```

We'll have to change it anyway once we implement `connect()` and `launchPersistentContext()`, which also take an `options` object.

Author: I've made the larger change so that the options for `launch` now live in its own page. I've not created pages for `connect` or `laucnPersistentContext` since they're not implemented but documented them as WIP.

Resolved in: 805dd25465b382f0e5edc1797adad4ac3000d8a1

> | Method            | Type     | Default | Description                                                                                                                                                                                                                                                           |

Reviewer: ```suggestion
| Property            | Type     | Default | Description                                                                                                                                                                                                                                                           |
```

| executablePath    | string   | `null`  | Override search for browser executable in favor of specified absolute path.                                                                                                                                                                                           |
| headless          | boolean  | `true`  | Show browser GUI or not.                                                                                                                                                                                                                                              |
| ignoreDefaultArgs | string[] | `null`  | Ignore any of the default arguments included when launching browser process.                                                                                                                                                                                          |

> | proxy             | string   | `null`  | Specify to set browser's proxy config.                                                                                                                                                                                                                                |

Reviewer: ```suggestion
| proxy             | string   | `null`  | Specify to set browser's proxy configuration.                                                                                                                                                                                                                                |
```
